Build using python limited API

mickael9 opened the pull request:
SWIG version 4.2.0 and up supports building using the Python limited API
This provides binary wheels with stable ABI that will work with any
python 3 version >= 3.4 (including future unreleased versions)

mickael9 committed:
Use macos-13 runner instead of macos-latest

macos-latest now uses arm64 runners which don't work with the current workflow.

mickael9 committed:
Upgrade deprecated node 16 actions

ghost commented:
@mickael9 Thanks, your fork is exactly what we needed, since our org requires Python >= 3.12 now. 

@jupe Have you seen this PR? It solves a lot of version update headaches I think and should make your life easier

jupe reviewed (approved):
need to pass CI before merging

jupe commented on .github/workflows/build.yml:
```diff
@@ -12,59 +12,50 @@ jobs:
-        os: [ ubuntu-20.04, macos-latest, windows-2019 ]
-        python-version: [ '3.8', '3.9', '3.10', '3.11' ]
+        os: [ ubuntu-20.04, macos-13, windows-2019 ]
+        python-version: [ '3.11' ]
```
we still want to run tests also with python 3.10

mickael9 replied:
If you are referring to the unit tests in `tox.ini`, they were not actually executed because the working directory was set incorrectly in the toxify step:
```
ROOT: No tox.ini or setup.cfg or pyproject.toml found, assuming empty tox.ini at /home/runner/work/pc-ble-driver-py/pc-ble-driver-py/pc-ble-driver
```

They would fail anyway since they require external hardware

I can change the python version to 3.10 if you prefer (the result is going to be the same)

jupe replied:
I means this build job. unit tests are different story.

mickael9 committed:
Build using python limited API

SWIG version 4.2.0 and up supports building using the Python limited API
This provides binary wheels with stable ABI that will work with any
python 3 version >= 3.4 (including future unreleased versions)
